src/api/valence-api.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const { LEVersion, LPVersion } = require('../constants');
+
+module.exports = class ValenceApi {
+	constructor(
+		valence,
+		fetch = require('node-fetch')
+	) {
+		this._valence = valence;
+		this._fetch = fetch;
+	}
+
+	async getContent(instanceUrl, orgUnit, module = null) {
+		const url = module
+			? new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/content/modules/${module.Id}/structure/`, instanceUrl)
+			: new URL(`/d2l/api/le/${LEVersion}/${orgUnit.Identifier}/content/root/`, instanceUrl);
+		const signedUrl = this._valence.createAuthenticatedUrl(url, 'GET');
+
+		const response = await this._fetch(signedUrl);
+
+		this._assertResponse(response);
+
+		return response.json();
```

Comment:
I think returning the entire response causes a lot of leakage of Valence concerns and details into the rest of the code.  I think it's worthwhile to return only the properties that we care about from all of these methods.

Reply:
This is now internal.